internal/dashboards/openshift/openshift.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openshift
+
+import (
+	"context"
+	_ "embed"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+//go:embed metrics-dashboard.json
+var dashboardContent string
+
+const (
+	openshiftDashboardsNamespace = "openshift-config-managed"
+)
+
+func CreateOpenShiftDashboard(clientset kubernetes.Interface) error {
+
+	cm := corev1.ConfigMap{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "opentelemetry-collector",
+			Namespace: openshiftDashboardsNamespace,
+			Labels: map[string]string{
+				"console.openshift.io/dashboard": "true",
+			},
+		},
+		Data: map[string]string{
+			"otel.json": dashboardContent,
+		},
+	}
```

Comment:
Why don't we follow our usual pattern for manifest generation here?

Reply:
As I mentioned in #2995:
> It is not necessary to create this dashboard per collector so, after a talk with @pavolloffay we think it would be nice if we create this dashboard as part of the steps to set up the operator.
> 
> We don't need to reconcile this `ConfigMap`.

If I'm not missing something, using the rules we use for the other ConfigMaps will add all the annotations and everything to reconcile the `ConfigMap`, right?